client/src/CheckinPage.tsx

```diff
@@ -0,0 +1,44 @@
+import Button from '@mui/material/Button';
+import React, { useState } from 'react';
+
+function Checkin_page() {
+    const deskId = document.location.search.substr(1);
+    const [currentDeskState, setCurrentDeskState] = useState('free');
+    fetch(`http://localhost:3001/api/desk/${deskId}`)
```

Comment:
deduplicate url to a const variable. 

the root of the URL should be in the .env, you will get e.g.: 

```
const URL = `${process.env.REACT_APP_ROOT_URL}/api/desk/${deskId}`;
```

Reply:
Also see https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env